lib/task/tools/settingsTask.class.php

```diff
@@ -0,0 +1,351 @@
+            }
+
+            // Clean up after test file write
+            fclose($fp);
+            unlink($options['to-file']);
+        }
+    }
+
+    /**
+     * @see sfBaseTask
+     */
+    protected function configure()
+    {
+        $this->addArguments([
+            new sfCommandArgument(
+                'operation',
+                sfCommandArgument::REQUIRED,
+                'Setting operation ("get" or "set").'
+            ),
+            new sfCommandArgument(
+                'name',
+                sfCommandArgument::REQUIRED,
+                'Name of setting.'
+            ),
+        ]);
+
+        $this->addOptions([
+            new sfCommandOption(
+                'application',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'The application name',
+                'qubit'
+            ),
+            new sfCommandOption(
+                'env',
+                null,
+                sfCommandOption::PARAMETER_REQUIRED,
+                'The environment',
+                'cli'
+            ),
+            new sfCommandOption(
+                'connection',
+                null,
+                sfCommandOption::PARAMETER_REQUIRED,
+                'The connection name',
+                'propel'
+            ),
+
+            // Tool options
+            new sfCommandOption(
+                'culture',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'Culture (default: "en"))',
+                'en'
+            ),
+            new sfCommandOption(
+                'scope',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'Scope of setting to get or set',
+                null
+            ),
+            new sfCommandOption(
+                'value',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'Value (if using "set" operator and not using the "from-file" option))',
+                null
+            ),
+            new sfCommandOption(
+                'from-file',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'File whose contents are to be used as a value (if using "set" operation)',
+                null
+            ),
+            new sfCommandOption(
+                'to-file',
+                null,
+                sfCommandOption::PARAMETER_OPTIONAL,
+                'File to write value to (if using "get" operation)',
+                null
+            ),
+            new sfCommandOption(
+                'strict',
+                null,
+                sfCommandOption::PARAMETER_NONE,
+                'Prevent creation of new settings when performing "set" operation',
+                null
+            ),
+        ]);
+
+        $this->namespace = 'tools';
+        $this->name = 'setting';
```

Comment:
Nitpick: I think it's better for the class name to be `SettingTask` or to make `$this->name = 'settings';` for consistency.

Reply:
Good call. Changing the name to "settings" makes sense.